source/common/router/config_impl.h

```diff
@@ -860,6 +872,21 @@ class RouteEntryImplBase : public RouteEntry,
     envoy::type::v3::FractionalPercent fractional_runtime_default_{};
   };
 
+  struct RouteStatsContext {
+    RouteStatsConfigSharedPtr route_stats_config_;
+    Stats::ScopeSharedPtr route_stats_scope_;
+    const Stats::StatNameManagedStorage& route_stat_name_storage_;
+
+  public:
+    RouteStatsContext(RouteStatsConfigSharedPtr route_stats_config,
```

Comment:
you don't really need a constructor here -- you can just initialize at call-sites right?

Reply:
It fails at call site while initializing with std::make_unique